app/src/main/java/io/apicurio/registry/store/RegistryStorage.java

```diff
@@ -0,0 +1,78 @@
+    
+    public void deleteArtifact(String artifactId) throws ArtifactNotFoundException, RegistryStorageException;
+    
+    public StoredArtifact getArtifact(String artifactId) throws ArtifactNotFoundException, RegistryStorageException;
+    
+    public ArtifactMetaData updateArtifact(String artifactId, String content) throws ArtifactNotFoundException, RegistryStorageException;
+    
+    public ArtifactMetaData getArtifactMetaData(String artifactId) throws ArtifactNotFoundException, RegistryStorageException;
+    
+    public void updateArtifactMetaData(String artifactId, EditableMetaData metaData) throws ArtifactNotFoundException, RegistryStorageException;
+    
+    public List<Rule> getArtifactRules(String artifactId) throws ArtifactNotFoundException, RegistryStorageException;
+    
+    public void createArtifactRule(String artifactId, Rule rule) throws ArtifactNotFoundException, ArtifactRuleAlreadyExistsException, RegistryStorageException;
+    
+    public void deleteArtifactRules(String artifactId) throws ArtifactNotFoundException, RegistryStorageException;
+    
+    public Rule getArtifactRule(String artifactId, String ruleName) throws ArtifactNotFoundException, RuleNotFoundException, RegistryStorageException;
+    
+    public void deleteArtifactRule(String artifactId, String ruleNam) throws ArtifactNotFoundException, RuleNotFoundException, RegistryStorageException;
+    
+    public List<Long> getArtifactVersions(String artifactId) throws ArtifactNotFoundException, RegistryStorageException;
+    
+    public StoredArtifact getArtifactVersion(String artifactId, long version) throws ArtifactNotFoundException, VersionNotFoundException, RegistryStorageException;
+    
+    public void deleteArtifactVersion(String artifactId, long version) throws ArtifactNotFoundException, VersionNotFoundException, RegistryStorageException;
+    
+    public VersionMetaData getArtifactVersionMetaData(String artifactId, long version) throws ArtifactNotFoundException, VersionNotFoundException, RegistryStorageException;
+    
+    public void updateArtifactVersionMetaData(String artifactId, long version, EditableMetaData metaData) throws ArtifactNotFoundException, VersionNotFoundException, RegistryStorageException;
+    
+    public void deleteArtifactVersionMetaData(String artifactId, long version) throws ArtifactNotFoundException, VersionNotFoundException, RegistryStorageException;
+    
+    public List<Rule> getGlobalRules() throws RegistryStorageException;
+    
+    public void createGlobalRule(Rule rule) throws RegistryStorageException;
+    
+    public void deleteGlobalRules() throws RegistryStorageException;
+    
+    public Rule getGlobalRule(String ruleName) throws RuleNotFoundException, RegistryStorageException;
+
+    public void updateGlobalRule(Rule rule) throws RuleNotFoundException, RegistryStorageException;
+
+    public void deleteGlobalRule(String ruleName) throws RuleNotFoundException, RegistryStorageException;
+
+}
```

Comment:
There is no `getArtifactVersionGlobal(long globalArtifactVersionId)`. The compatibility requires that artifact versions have a globally unique ID.

Reply:
Hmmm, I guess that could be "hidden" into artifactId being a number, and translate that as global id ... ?

Reply:
Which endpoint in the Confluent API allows this?

Reply:
`
  @GET
  @Path("/ids/{id}")
  @ApiOperation("Get the schema string identified by the input ID.")
  @ApiResponses(value = {
      @ApiResponse(code = 404, message = "Error code 40403 -- Schema not found\n"),
      @ApiResponse(code = 500, message = "Error code 50001 -- Error in the backend data store\n")})
  @PerformanceMetric("schemas.ids.get-schema")
  public SchemaString getSchema(
      @ApiParam(value = "Globally unique identifier of the schema", required = true)
      @PathParam("id") Integer id) {
`

Reply:
We have to have 2 global IDs for the same data: `long globalArtifactVersionId` and tuple `[String artifactId, long version]`. The problem is that we cannot translate between them directly @alesj because `String artifactId` can't be globally unique for versions and `String artifactId` can't fit into `long globalArtifactVersionId`.

Reply:
OK I've re-read the confluent registry API docs and now I understand.  I didn't quite understand this before.

Reply:
I think this changes some core concepts a bit.  I didn't realize that each **version** of a schema in confluent had a globally unique ID.  And in fact, confluent doesn't have a unique server-generated ID for the **artifact** - just the version.  I think that simplifies things for us a bit.  I think that our `artifactId` can remain a string and it can be **optionally** client-provided.  If not provided then we can generate a UUID (separate from the storage layer).  Either way, we'll return the `artifactId` when creating an artifact.  We'll also return the versionId, which is always a server-generated globally unique `long`.  But that `long` value is the unique ID of the **version** of the artifact. 

Does that sound reasonable to you guys, @jsenko and @alesj ?

Reply:
This also means we don't need to restrict the client-provided ID and we don't need any sort of tuple or anything.  The call to `createArtifact` in the storage layer will return an object containing relevant meta-data such as the generated `versionId`.

Reply:
So, to understand correctly, **versionId** is this global unique ID @jsenko used before / mentioned now?

Reply:
What I thought was, when we have a "getArtifact(String artifactId)" we could try and parse artifactId into number -- if we succeed, then this is global unique ID / versionId.